lib/internal/test_runner/harness.js

```diff
@@ -113,19 +122,45 @@ function setup(root) {
 
   root.reporter.pipe(process.stdout);
   root.reporter.version();
+
+  wasRootSetup = true;
+  return root;
 }
 
 function test(name, options, fn) {
-  // If this is the first test encountered, bootstrap the test harness.
-  if (this.subtests.length === 0) {
-    setup(this);
-  }
+  const subtest = setup(root).createSubtest(name, options, fn);
+  return subtest.start();
+}
 
-  const subtest = this.createSubtest(name, options, fn);
+function getParent() {
+  return testResources.get(executionAsyncId()) || setup(root);
+}
 
-  return subtest.start();
+function describe(name, options, fn, overrides) {
+  const parent = getParent();
+  const suite = parent.createSubSuite(name, options, fn, overrides);
+  if (parent === root) {
+    suite.run();
+  }
+  return suite;
 }
 
-const root = new Test({ name: '<root>' });
+function it(name, options, fn, overrides) {
+  return getParent().createSubtest(name, options, fn, overrides);
+}
+
+function enrichMethod(method) {
+  ['skip', 'only'].forEach((keyword) => {
+    method[keyword] = function(name, options, fn) {
```

Comment:
```suggestion
    method[keyword] = (name, options, fn) => {
```
so it can't be `new`ed.